resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -164,6 +165,9 @@ private[spark] class Client(
         }
       } finally {
         kubernetesResourceCleaner.deleteAllRegisteredResourcesFromKubernetes(kubernetesClient)
+        if (loggingWatch.nonEmpty) {
+          loggingWatch.get.shutdown()
```

Comment:
Hm, I think we should prefer fixing what's upstream as opposed to adding to our code.